test/extended/util/url/url.go

```diff
@@ -362,7 +363,18 @@ func (ut *Test) ToShell(i int) string {
 			post = post + " -d '' "
 		}
 	}
-	cmd := fmt.Sprintf(`curl -X %s %s %s -s -S -o /tmp/body -D /tmp/headers %q`, ut.Req.Method, strings.Join(headers, " "), post, ut.Req.URL)
+	resolve := ""
+	if len(ut.ProxyHost) != 0 {
+		host, port, err := net.SplitHostPort(ut.Req.URL.Host)
+		if err != nil {
+			host, port = ut.Req.URL.Host, "80"
+			if ut.Req.URL.Scheme == "https" {
+				port = "443"
+			}
+		}
+		resolve = fmt.Sprintf("--resolve %s:%s:%s", host, port, ut.ProxyHost)
```

Comment:
will this function on ipv6?

Reply:
It should, if the address is bracketed.  I've added a comment to `Through` to mention that IPv6 addresses should be bracketed.  I've also replaced `--resolve` with `--connect-to`, which handles IPv4 addresses, bracketed IPv6 addresses, and host names (the last would have failed with `--resolve`, which would cause problems on platforms such as AWS where the LB status has a host name rather than an IP address).